cookbook/cache/form_csrf_caching.rst

```diff
@@ -0,0 +1,54 @@
+.. index::
+    single: Cache; CSRF; Forms
+
+Caching Pages that Contain CSRF Protected Forms
+===============================================
+
+CSRF - or `Cross-site request forgery`_ - is a method by which a malicious
+user attempts to make your legitimate users unknowingly submit data that
+they don't intend to submit. Fortunately, CSRF attacks can be prevented by
+using a CSRF token inside your forms.
+
+For more information about how this protection works in Symfony, please
+check :ref:`CSRF Protection <forms-csrf>`.
+
+.. note::
+
+    A `Security CSRF Component`_ was introduced in Symfony 2.4 which provides
+    a class CsrfTokenManager for generating and validating CSRF tokens
+
+Why Varnish does not cache by default these pages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are many ways to generate unique tokens for each user but in order get
+them validated when the form is submitted, you need to store them inside the
+PHP Session.
+
+If you are using Varnish or some similar reverse proxy cache and you try to cache
+pages containing forms with CSRF token protection, you will see that, by default,
+Varnish fails to cache them.
```

Comment:
nope, when i read https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php#L447-L450 we call forward which checks whether the response is cacheable and sets it to private otherwise. this is a bit weird - i guess its a safety precaution to fix backends that accidentally try to make a response that was generated with cookies public. if this was indeed intended behaviour of the backend, they would have to change the option - but that is not the point here.

i am convinced that this code does NOT cache a response if the request had any cookie in it.

in the text above i would only mention varnish the first time and end with `you will see that, by default, the reverse proxy refuses to cache.`